### PR TITLE
OSDOCS-1172: Adding docs for token inactivity timeout

### PR DESCRIPTION
**Pull request by bergerhoffer**
https://issues.redhat.com/browse/OSDOCS-1172

Previews (internal): 
* http://file.rdu.redhat.com/~ahoffer/2020/OSDOCS-1172/authentication/configuring-internal-oauth.html#oauth-token-inactivity-timeout_configuring-internal-oauth
* http://file.rdu.redhat.com/~ahoffer/2020/OSDOCS-1172/authentication/configuring-oauth-clients.html

### CODE REVIEW

File: authentication/configuring-internal-oauth.adoc [hunk 1]
```diff
@@ -4,6 +4,7 @@ include::modules/common-attributes.adoc[]
 :context: configuring-internal-oauth
 toc::[]
 
+// TODO: What on earth is this?
 [IMPORTANT]
 ====
 Configuring these options must change because they're set in the master
```
Comment (bergerhoffer) [inline authentication/configuring-internal-oauth.adoc:10]: @kalexand-rh Any chance you remember what this statement is/was supposed to mean? Looks like it was added here: https://github.com/openshift/openshift-docs/pull/13051/files#diff-9b330ab3e6f94100d5ff06d0a7a83d6cR8-R12. But I can't track down any further info on it from the related Jiras.

Almost sounds like some internal note that shouldn't have been rendered, but I'm not sure. Let me know if you think it can be removed or if there's a reason we need it.

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+. Verify that the token inactivity timeout was successfully set.
+
+.. Log in to the cluster with an identity from your password-based IDP. Be sure to use the OAuth client that you just configured.
+
+.. Perform an action and verify that it was successful.
+
+.. Wait longer than the configured timeout without using the identity. In this procedure's example, wait longer than 10 minutes.
+
+.. Attempt to perform an action from the same identity's session.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout-clients.adoc:55]: s/Attempt to/Try to/ ?

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
+// Module included in the following assemblies:
+//
+// * authentication/understanding-internal-oauth.adoc
+
+[id="oauth-token-inactivity-timeout_{context}"]
+= Configuring token inactivity timeout for the internal OAuth server
+
+You can configure OAuth tokens to expire after a certain amount of time that they have been inactive. By default, there is no token inactivity timeout set.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:8]: Same comment as above about "time that they have been inactive."

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+[NOTE]
+====
+If token inactivity timeout is also configured in your OAuth client, that value overrides the timeout set in the internal OAuth server configuration.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:12]: _Same comment as above_
For consideration:
s/If token inactivity/If the token inactivity/
s/the timeout set in the/the timeout that is set in the/

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+====
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:17]: See previous comment and question about cluster administrator wording

**Comment by bergerhoffer** [comment]
@vareti Can you please review?

Previews (internal):
* http://file.rdu.redhat.com/~ahoffer/2020/OSDOCS-1172/authentication/configuring-internal-oauth.html#oauth-token-inactivity-timeout_configuring-internal-oauth
* http://file.rdu.redhat.com/~ahoffer/2020/OSDOCS-1172/authentication/configuring-oauth-clients.html

Note that I created a whole new section for configuring OAuth clients (second link above) and moved some existing content there, in addition to the timeout procedure. Please give that a check too to make sure it makes sense as I have it.

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+...
+spec:
+  tokenConfig:
+    accessTokenInactivityTimeout: 400s <1>
+----
+<1> Set a value with the appropriate units, for example `400s` for 400 seconds, or `30m` for 30 minutes. The minimum allowed timeout value is `300s`.
+
+.. Save the file to apply the changes.
+
+. Wait for the OAuth server Pods to restart:
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:47]: For these two steps, can you clarify what action they are taking when entering `oc get`? Maybe something like:

> . Check that the OAuth server pods have restarted:

(It's a little confusing that the task is to "wait" followed by running a command.)

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+.Example output
+[source,terminal]
+----
+NAME             VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
+kube-apiserver   4.6.0     True        False         False      145m
+----
++
+If `PROGRESSING` is showing `True`, wait a few minutes and try again.
+
+. Verify that the token inactivity timeout was successfully set.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:83]: Similar to previous comment, my preference would be to remove `successfully` here.

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+.. Log in to the cluster with an identity from your password-based IDP.
+
+.. Execute a command and verify that it was successful.
+
+.. Wait longer than the configured timeout without using the identity. In this procedure's example, wait longer than five minutes.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:89]: Perhaps "wait longer than 300 seconds"? But in the sample spec, you show `400s`. 

Also, my understanding with ISG is that you would write it as "5 minutes" (use a numeral with numbers combined with a unit of measurement, from p. 157).

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+.. Attempt to execute a command from the same identity's session.
```
Comment (bobfuru) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:91]: same as above, my personal pref would be "Try" vs. "Attempt" in this step.

File: modules/oauth-default-clients.adoc [hunk 10]
```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// * authentication/configuring-oauth-clients.adoc
+
+[id="oauth-default-clients_{context}"]
+= Default OAuth clients
+
+The following OAuth clients are automatically created when starting the {product-title} API:
```
Comment (bobfuru) [inline modules/oauth-default-clients.adoc:8]: same as above, I don't think you need to say "automatically" (esp. since the API suggests it is done programmatically anyway)

File: modules/oauth-default-clients.adoc [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+|`openshift-browser-client`
+|Requests tokens at `<namespace_route>/oauth/token/request` with a user-agent that can handle interactive logins. ^[1]^
+
+|`openshift-challenging-client`
+|Requests tokens with a user-agent that can handle `WWW-Authenticate` challenges.
+
+|===
+[.small]
+--
+1. `<namespace_route>` refers to the namespace's route. This is found by
```
Comment (bobfuru) [inline modules/oauth-default-clients.adoc:24]: Can you remove the apostrophe and still retain the meaning? It's kind of hard to read "namespace's route". Or maybe "the route of the namespace", though that's a bit wordy. ¯\_(ツ)_/¯

File: modules/oauth-token-requests.adoc [hunk 12]
```diff
@@ -12,38 +12,37 @@ created when starting the {product-title} API:
...
 |===
-
-[NOTE]
-====
-_<namespace_route>_ refers to the namespace's route. This is found by
-running the following command.
-
+[.small]
+--
+1. `<namespace_route>` refers to the namespace's route. This is found by
```
Comment (bobfuru) [inline modules/oauth-token-requests.adoc:26]: Same question about `namespace's` here.

**Review by bobfuru — COMMENTED** [review]
I have a bunch of meddling questions about style and usage. Most of these can be ignored or overridden of course. But if you'd like me to have another look or want any more clarification, please let me know.

**Comment by bergerhoffer** [comment]
Thanks a lot for the great feedback @bobfuru! I'll address the feedback once I'm back from PTO!

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+.Example output
+[source,terminal]
+----
+NAME             VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
+authentication   4.6.0     True        False         False      145m
+----
+
+. Wait for a new revision of the Kubernetes API server Pods to roll out. This will take several minutes.
++
+// TODO: Note that if you do this quickly enough, it will say False before it turns to True, then goes to False after ~10 minutes.
```
[the remark this answers is not in the document]
Comment (vareti) [reply modules/oauth-configuring-token-inactivity-timeout.adoc:65]: @bergerhoffer the reconcile logic runs periodically and it could be the reason for the behavior you are seeing.

File: modules/oauth-default-clients.adoc (hunk 11 above)
Comment (bergerhoffer) [reply modules/oauth-default-clients.adoc:24]: Agree - this was copied from elsewhere so I will change it there too.

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+You can configure OAuth clients to expire OAuth tokens after a certain amount of time that they have been inactive. By default, there is no token inactivity timeout set.
+
+[NOTE]
+====
+If token inactivity timeout is also configured in the internal OAuth server configuration, the timeout set in the OAuth client overrides that value.
+====
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply modules/oauth-configuring-token-inactivity-timeout-clients.adoc:17]: @bobfuru So I use this same phrasing in all of my procedures (that it's required for). But I'm not opposed to adopting a standard wording if you want to bring it up to the team?

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+
+.. Save the file to apply the changes.
+
+. Verify that the token inactivity timeout was successfully set.
+
+.. Log in to the cluster with an identity from your password-based IDP. Be sure to use the OAuth client that you just configured.
+
+.. Perform an action and verify that it was successful.
+
+.. Wait longer than the configured timeout without using the identity. In this procedure's example, wait longer than 10 minutes.
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply modules/oauth-configuring-token-inactivity-timeout-clients.adoc:53]: Updated to 600 seconds so that they can easily match that up to the config without breaking out a calculator ;)

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+
+. Update the OAuth client configuration to set a token inactivity timeout.
+
+.. Edit the OAuth client configuration:
++
+[source,terminal]
+----
+$ oc edit oauthclient <oauth_client> <1>
+----
+<1> Replace `<oauth_client>` with the OAuth client to configure, for example, `console`.
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply modules/oauth-configuring-token-inactivity-timeout-clients.adoc:30]: I don't think we have a hard and fast rule about it, but I do think that the IBM style guide encourages calling out when values should be replaced. I've been trying to do it more often, though I don't typically do it for super obvious ones. Could be worthwhile to bring up to the team though.

File: modules/oauth-configuring-token-inactivity-timeout.adoc (hunk 6 above)
Comment (bergerhoffer) [reply modules/oauth-configuring-token-inactivity-timeout.adoc:47]: Good point, thanks!

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (bobfuru) [reply modules/oauth-configuring-token-inactivity-timeout-clients.adoc:17]: That makes sense. Let me try to research different occurrences. I'd like to understand if "you are a cluster administrator" is analogous to "you have the `cluster-admin` role" or if there are other permutations. And good idea, I can bring it up with the team.

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+
+[NOTE]
+====
+If the token inactivity timeout is also configured in the internal OAuth server configuration, the timeout that is set in the OAuth client overrides that value.
+====
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have configured a password-based identity provider (IDP), such as HTPasswd or LDAP.
```
Comment (vareti) [inline modules/oauth-configuring-token-inactivity-timeout-clients.adoc:18]: It doesn't have to be a password based IDP. Any IDP should work with this feature.

File: modules/oauth-configuring-token-inactivity-timeout.adoc [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+
+[NOTE]
+====
+If the token inactivity timeout is also configured in your OAuth client, that value overrides the timeout that is set in the internal OAuth server configuration.
+====
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have configured a password-based identity provider (IDP), such as HTPasswd or LDAP.
```
Comment (vareti) [inline modules/oauth-configuring-token-inactivity-timeout.adoc:18]: same as above. token timeout works with any idp

File: modules/oauth-configuring-token-inactivity-timeout-clients.adoc (hunk 17 above)
Comment (bergerhoffer) [reply modules/oauth-configuring-token-inactivity-timeout-clients.adoc]: Updated.

File: modules/oauth-configuring-token-inactivity-timeout.adoc (hunk 18 above)
Comment (bergerhoffer) [reply modules/oauth-configuring-token-inactivity-timeout.adoc]: Updated.

**Comment by bergerhoffer** [comment]
@rhpmali Can you please QE review these docs? Thanks!

Previews: 
* https://osdocs-1172--ocpdocs.netlify.app/openshift-enterprise/latest/authentication/configuring-internal-oauth.html#oauth-token-inactivity-timeout_configuring-internal-oauth
* https://osdocs-1172--ocpdocs.netlify.app/openshift-enterprise/latest/authentication/configuring-oauth-clients.html

File: authentication/configuring-internal-oauth.adoc (hunk 1 above)
Comment (bergerhoffer) [reply authentication/configuring-internal-oauth.adoc:10]: I'm going to remove this note. I think @kalexand-rh is out for a bit so we can always add it back in if she confirms that it's needed. @rhpmali Let me know if you have any thoughts on this note or objections to its removal.
